cyclonedds-cxx master is not compatible with cyclonedds master

## Required information

**Operating system:**
Ubuntu 18.04 LTS

**Cyclone Version**

  -  cyclonedds - master branch, eclipse-cyclonedds/cyclonedds@b74b2f9c5e009cb025a7aac376792d58ad4700df
   - cyclonedds-cxx - template_streaming , 212146f849f93e5dd5980ce60999e58c8ed17a0e

**Observed result or behaviour:**
Compiling the master of `cyclonedds-cxx` with the master of `cyclonedds` leads to the following compilation error

```sh
/home/sumanth.nirmal/apex_middleware_ws/x86_64/cyclonedds-cxx/src/idlcxx/src/streamers.c:722:44: error: ‘IDL_TRUE’ undeclared (first use in this function); did you mean ‘IDL_TYPE’?
       ((const idl_member_t *)node)->key == IDL_TRUE)
                                            ^~~~~~~~
                                            IDL_TYPE
/home/sumanth.nirmal/apex_middleware_ws/x86_64/cyclonedds-cxx/src/idlcxx/src/streamers.c:722:44: note: each undeclared identifier is reported only once for each function it appears in
/home/sumanth.nirmal/apex_middleware_ws/x86_64/cyclonedds-cxx/src/idlcxx/src/streamers.c: In function ‘process_switch_type_spec’:
/home/sumanth.nirmal/apex_middleware_ws/x86_64/cyclonedds-cxx/src/idlcxx/src/streamers.c:1132:32: error: ‘IDL_TRUE’ undeclared (first use in this function); did you mean ‘IDL_TYPE’?
   if (switch_type_spec->key != IDL_TRUE)
                                ^~~~~~~~
                                IDL_TYPE
/home/sumanth.nirmal/apex_middleware_ws/x86_64/cyclonedds-cxx/src/idlcxx/src/types.c: In function ‘emit_enum’:
/home/sumanth.nirmal/apex_middleware_ws/x86_64/cyclonedds-cxx/src/idlcxx/src/types.c:343:11: error: incompatible types when assigning to type ‘uint32_t {aka unsigned int}’ from type ‘const struct <anonymous>’
     value = enumerator->value;
           ^
src/idlcxx/CMakeFiles/idlcxx.dir/build.make:82: recipe for target 'src/idlcxx/CMakeFiles/idlcxx.dir/src/types.c.o' failed
```

**Expected result or behaviour:**

The master branches of `cyclonedds` and `cyclonedds-cxx` should be compatible and can be compiled without any errors.

**Other Info**

The master on `cyclonedds-cxx` is probably broken because of the changes in this MR https://github.com/eclipse-cyclonedds/cyclonedds/pull/923

--- a/src/idlcxx/src/streamers.c
+++ b/src/idlcxx/src/streamers.c
@@ -719,7 +719,7 @@ process_member(
   type_spec = ((const idl_member_t *)node)->type_spec;
   /* only use the @key annotations when you do not use the keylist */
   if (!(pstate->flags & IDL_FLAG_KEYLIST) &&
-      ((const idl_member_t *)node)->key == IDL_TRUE)
+      ((const idl_member_t *)node)->key.value)
     loc.type |= KEY_INSTANCE;
 
   IDL_FOREACH(declarator, ((const idl_member_t *)node)->declarators) {
@@ -1129,7 +1129,7 @@ process_switch_type_spec(
     return IDL_RETCODE_NO_MEMORY;
 
   /* short-circuit if switch type specifier is not a key */
-  if (switch_type_spec->key != IDL_TRUE)
+  if (!switch_type_spec->key.value)
     return IDL_RETCODE_OK;
 
   if (putf(&streams->key_write, writefmt)

--- a/src/idlcxx/src/types.c
+++ b/src/idlcxx/src/types.c
@@ -340,7 +340,7 @@ emit_enum(
 
   IDL_FOREACH(enumerator, _enum->enumerators) {
     name = get_cpp11_name(enumerator);
-    value = enumerator->value;
+    value = enumerator->value.value;
     fmt = (value == skip) ? "%s%s" : "%s%s = %" PRIu32 "\n";
     if (idl_fprintf(gen->header.handle, fmt, sep, name, value) < 0)
       return IDL_RETCODE_NO_MEMORY;

